proj/hps_accel/src/tensorflow/lite/kernels/internal/reference/integer_ops/fully_connected.h

```diff
@@ -93,6 +95,20 @@ inline void FullyConnected(
   }
 }
 
+inline void FullyConnectedWithPackedInt4Weights(
+    const FullyConnectedParams& params, const RuntimeShape& input_shape,
+    const int8_t* input_data, const RuntimeShape& filter_shape,
+    const int8_t* filter_data, int8_t* unpacked_filter_data,
+    const RuntimeShape& bias_shape, const int32_t* bias_data,
+    const RuntimeShape& output_shape, int8_t* output_data) {
+  TFLITE_DCHECK_NE(unpacked_filter_data, nullptr);
+  tflite::tensor_utils::UnpackDenseInt4IntoInt8(
+      filter_data, filter_shape.FlatSize(), unpacked_filter_data);
+  FullyConnected(params, input_shape, input_data, filter_shape,
+                 unpacked_filter_data, bias_shape, bias_data, output_shape,
+                 output_data);
+}
+
```

Comment:
Where was this before/why did we have to move it local code?

Reply:
This file exists in TFLM.   This is a locally modified version.   What it *should* be is a copy of the current TFLM file, patched with our local changes.   Instead, we maintain a copy of the entire file, which then overlays the TFLM version.   When modifications occur upstream, we need to manually copy them into this copy of the file.

Essentially, I'm manually propagating this change: https://github.com/tensorflow/tflite-micro/commit/1d1d6b968d03c289da6ff9c8384df22aac29391f#diff-2cb25dbe2523a0754af5b5057b800add3642578751a4083dd094b5b99ecdecb2  into our modified copy.

Again, a better way would be to store just the patches for our local changes, so that they could be applied to the latest TFLM version.   That way, most of the time we could automatically combine upstream changes with our local changes (except when there is a conflict).   But that would be a bit more complicated to set up.

Reply:
Thanks for explaining! Got it now.